### ISSUE TITLE
Incorrect hash: sharderKeys used in place of mpkKeys while calculating magic block hash

### ISSUE THREAD

**Issue by bbist** [0chain/0chain#146]
In file **0chain.net/chaincore/block/magic_block_entity.go**, method **GetHashBytes** uses value **sharderKeys** instead of **mpkKeys** in line 85, to iterate over and append to data variable. It seems like a bug.
```
func (mb *MagicBlock) GetHashBytes() []byte {
        ...
        ...
	sort.Strings(mpkKeys)
->	for _, v := range sharderKeys {
		data = append(data, []byte(v)...)
	}
	data = append(data, []byte(strconv.Itoa(mb.T))...)
	data = append(data, []byte(strconv.Itoa(mb.N))...)
	return encryption.RawHash(data)
}
```

**Comment by bbist** [comment]
Here's the reference to the change: https://github.com/0chain/0chain/commit/27199b9bcbd1071139d061628ab12032119924ab

**Comment by guruhubb** [comment]
@peterlimg is this closed?

**Comment by peterlimg** [comment]
yes, this is closed by #147
